dbms/src/Encryption/KeyspacesKeyManager.h

```diff
@@ -17,21 +17,33 @@
 #include <Common/Exception.h>
 #include <Common/LRUCache.h>
 #include <Common/TiFlashException.h>
+#include <Debug/MockProxyEncryptionFFI.h>
 #include <Encryption/KeyManager.h>
 #include <Encryption/MasterKey.h>
+#include <Storages/KVStore/FFI/ProxyFFI.h>
 #include <Storages/Page/PageStorage_fwd.h>
 
 namespace DB
 {
 
-struct EngineStoreServerWrap;
-
 /// KeyspacesKeyManager is a KeyManager implementation that designed for Cloud Storage Engine.
 /// It will store all encryption keys which have been encrypted using MasterKey in PageStorage.
 class KeyspacesKeyManager : public KeyManager
 {
 public:
-    explicit KeyspacesKeyManager(EngineStoreServerWrap * tiflash_instance_wrap_);
+#ifndef DBMS_PUBLIC_GTEST
```

Comment:
Make it `TiFlashRaftProxyHelper` and `MockProxyEncryptionFFI` as a template parameter, and remove the `#ifdef`?

Reply:
Looks like do not have obvious advantages, I prefer `#ifndef DBMS_PUBLIC_GTEST`.

Reply:
I am not quite against this. Actually I have used this in https://github.com/pingcap/tiflash/pull/8472, and @JaySon-Huang suggested me to use another solution. So I am not sure if it is OK here... Maybe also refer to him for advise?

Reply:
Using `#ifndef DBMS_PUBLIC_GTEST ...` to declare different types under "test mode" and "release mode" actually is a hack. Especially when we use it to define different class under different situation. It can easily being broken, for example:

* If we change the method signature of `TiFlashRaftProxyHelper`, it may be able to be built under CI but not be built under release mode.

I prefer using template to declare different class.

Reply:
ok, change to template.